### ISSUE TITLE
Pre-training compact backbones

### ISSUE THREAD

**Issue by goutamyg** [RobinWu218/SimDINO#37]
Hi @RobinWu218 @ultranity 

thank you for publishing the code and pre-trained models. Have you tried with compact backbones such as ViT-tiny (pure transformer-based), convnextv2-atto (pure CNN-based), or MobileViT (CNN + transformer hybrid) in your experiments? It will be interesting to know whether pre-training with compact backbone architectures require specific hyperparameter tuning. 

Also, the ablation study in the paper on the effect of batch size using ViT-small as the backbone. Can you please release the corresponding model weights as well? Thank you! 

Goutam

**Comment by RobinWu218** [comment]
1. We have not trained other models, but I can imagine they might require different optimization hyperparams(e.g. lr) due to the change in backbone architecture. 

2. Which models are you interested in? Also, the only differences are the batch sizes and I am curious about if you have specific use cases for these models in the ablation.

**Comment by dgcnz** [comment]
+1 for SimDINO ViT-S/16 (Table 6)

**Comment by goutamyg** [comment]
@RobinWu218 Thank you for your reply. 

1. The pre-training style of SimDINO (as well as DINO and DINOv2) seem specific to ViT-like architectures. The concept of CLS token does not exist in pure CNN-based (ConvNext/ConvNextV2) or many CNN + Transformer hybrid architecutres (e.g., MobileViT), which makes the pre-training scheme non-trivial to generalize to other architectures. From my understanding, the same argument holds true for patch-based masking operation as well. Please correct me if I am wrong. 

2. I am interested in the best performing ViT-S/16 from Table-6 in the paper.

**Comment by RobinWu218** [comment]
1. For the patch-feature reconstruction loss (in SimDINOv2), your statement is correct. If you are just using the original DINO/SimDINO, then it should not be specific to ViT-like architectures. In fact, the original DINO paper tried it on CNNs as well. As long as the model outputs a feature vector for an image, you can use the objective. CLS token is just one way to achieve this but other networks can do this as well.

2. Got it. We will upload the model checkpoint for ViT-S. We will likely retrain it for better performance since the ablation experiment uses simplistic configurations without many training epochs.

**Comment by goutamyg** [comment]
@RobinWu218 Thank you. I will close the ticket once the ViT-S models are uploaded. 

If possible, please upload the config file used anf logs for ViT-S training, so that we get an idea on what parameters were finetuned and loss values for a smaller architecture.

**Comment by goutamyg** [comment]
@RobinWu218 A gentle reminder to release ViT-S checkpoint. Thank you

**Comment by ultranity** [comment]
Initial ViT-S/16 ckpts for V2 branch, 100 epochs:

main config changes: smaller drop path rate(dpr), and below is the ckpt trained with drop path rate=0.1

vits16_reg4_DINOv2_dpr1_100ep
https://drive.google.com/file/d/1uIumF45WVIxTCVQQdo8c4QDISIGHMTrK/view?usp=drive_link
knn/linear prob: 73.088/74.628


vits16_reg4_SimDINOv2_dpr1_100ep
https://drive.google.com/file/d/1eRMkJVIB6FDH8R80zBMj0S7e99WIaHdB/view?usp=drive_link
knn/linear prob: 73.908/76.040

 drop path rate(dpr) 0.05/0.1/0.2/0.3 abblations:
| **config\KNN**                 | **10ep** | **20ep** | **30ep** | **40ep** | **50ep** | **60ep** | **70ep** | **80ep** | **90ep** | **100ep(knn/linear)** |
|----------------------------|--------|--------|--------|--------|--------|--------|--------|--------|--------|-----------------------|
| **vits16_dinov2_dpr05**    | 36.606 | 54.302 | 62.032 | 65.398 | 67.198 | 68.618 | 69.914 | 71.456 | 72.424 | 72.508/74.618         |
| **vits16_dinov2_dpr1**     | 40.376 | 58.476 | 64.666 | 66.790 | 68.134 | 69.520 | 70.916 | 72.264 | 72.982 | 73.088/74.628         |
| **vits16_dinov2_dpr2**     | 38.472 | 56.230 | 63.046 | 65.432 | 67.010 | 68.008 | 69.398 | 70.956 | 71.924 | 71.998/73.158         |
| **vits16_dinov2_dpr3**     | 37.882 | 55.918 | 62.306 | 64.098 | 65.220 | 66.120 | 67.596 | 68.988 | 69.890 | 70.072/71.082         |
| **vits16_simdinov2_dpr05** | 42.924 | 55.232 | 59.878 | 62.408 | 64.226 | 66.892 | 69.072 | 71.554 | 73.354 | 73.716/75.998         |
| **vits16_simdinov2_dpr1**  | 44.314 | 56.226 | 61.218 | 63.540 | 65.450 | 67.288 | 69.560 | 71.854 | 73.642 | 73.908/76.040         |
| **vits16_simdinov2_dpr2**  | 45.348 | 58.220 | 62.084 | 63.436 | 64.466 | 66.100 | 68.222 | 70.326 | 72.060 | 72.432/74.488         |
| **vits16_simdinov2_dpr3**  | 43.596 | 56.160 | 59.980 | 61.016 | 61.938 | 63.644 | 65.592 | 67.960 | 69.494 | 69.884/72.348         |
